app/models/dns-record.server.ts

```diff
@@ -33,12 +33,25 @@ export function getDnsRecordById(id: DnsRecord['id']) {
   return prisma.dnsRecord.findUnique({ where: { id } });
 }
 
-export function getUserDnsRecordCount(username: DnsRecord['username']) {
+const acmeChallengeRecordCount = (username: DnsRecord['username']) => {
   return prisma.dnsRecord.count({
     where: {
       username,
+      type: DnsRecordType.TXT,
+      subdomain: '_acme-challenge',
     },
   });
+};
+
+export async function getUserDnsRecordCount(username: DnsRecord['username']) {
+  let userRecordCount = await prisma.dnsRecord.count({
+    where: {
```

Comment:
Instead of 2 functions for count, we add a boolean that is default `true` to the function, maybe `excludeAcmeRecords` and use it to set `NOT` on the `where`

https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#not-1

Reply:
Thank you for the recommendation. I updated my PR!